doc/source/text.rst

```diff
@@ -205,6 +217,52 @@ and optional groups like
 
 can also be used.
 
+When ``str.extract`` method is used on an ``Index``, it returns a
+``DataFrame`` with as many columns as capture groups.
```

Comment:
we put it on the doc-string, yes? I think it should be mentioned in text.rst (maybe in a ``::note``) as I imagine this could be a bit surprising / not obvious to a user.